src/lgdo/__init__.py

```diff
@@ -66,11 +67,13 @@
     "VectorOfVectors",
     "VectorOfEncodedVectors",
     "WaveformTable",
-    "LH5Iterator",
-    "LH5Store",
+    "Iterator",
+    "Store",
```

Comment:
I guess we should remove these imports from the `lh5` submodule, especially  if we are dropping the `LH5` prefix. The user should `import lgdo.lh5.Store`, and not `lgdo.Store`. But see my comment about this prefix in the review summary.